apps/vscode/src/main.ts

```diff
@@ -125,9 +126,44 @@ export async function activate(context: vscode.ExtensionContext) {
 
   // activate providers common to browser/node
   activateCommon(context, host, engine, commands);
+
+  // if positron
+  setEditorOpener();
+
+  vscode.workspace.onDidChangeConfiguration(async (event) => {
+    if (event.affectsConfiguration('quarto.defaultEditor')) {
+      setEditorOpener();
+    }
+  });
+
+  const documentOpenHandler = vscode.window.onDidChangeActiveTextEditor(async () => {
+    // Check if the document language is "quarto"
+
```

Comment:
It seems like we don't have things quite right in this changing active text editor listener

1. A diff editor view of a quarto document immediately closes


https://github.com/user-attachments/assets/05a10155-8061-43b6-a97e-e970ea5ee6eb



2. I can't seem to go from Visual Mode to Source Mode explicitly

https://github.com/user-attachments/assets/bd14536d-08a1-48d0-8f62-248ab2b05280